### PR TITLE
Run java aarch64 tests on the CI (with an emulator)

### PR DESCRIPTION
**Pull request by jtattermusch**
Based on https://github.com/protocolbuffers/protobuf/pull/8479 (only the last few commits need to be reviewed).

- build protoc under crosscompiler
- then build & tests java code under qemu emulator.

Example passing adhoc run: http://sponge2.corp.google.com/95eaabd8-a929-4dfd-839a-4b4bef078391

### CODE REVIEW

**Comment by jtattermusch** [comment]
@dlj-NaN  please review or assign to the right reviewer for protobuf java.

File: kokoro/linux/aarch64/python_crosscompile_aarch64.sh [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
 
 ./autogen.sh
-CXXFLAGS="-fPIC -g -O2" ./configure
+CXXFLAGS="-fPIC -g -O2" ./configure --host=aarch64
 make -j8
 
+# create a simple shell wrapper that runs crosscompiled protoc under qemu
+echo '#!/bin/bash' >protoc_qemu_wrapper.sh
+echo 'exec qemu-aarch64 "../src/protoc" "$@"' >>protoc_qemu_wrapper.sh
+chmod ugo+x protoc_qemu_wrapper.sh
```
Comment (fowles) [inline kokoro/linux/aarch64/python_crosscompile_aarch64.sh:18]: pull request says java, but this was also part of your python PR...

File: kokoro/linux/java_aarch64/build.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+#
+# This is the top-level script we give to Kokoro as the entry point for
+# running the "continuous" and "presubmit" jobs.
+
+set -ex
+
+# Change to repo root
+cd $(dirname $0)/../../..
```
Comment (fowles) [inline kokoro/linux/java_aarch64/build.sh:9]: cd $(git rev-parse --show-toplevel)

seems more reliable than counting directories

File: kokoro/linux/python_aarch64/build.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+#
+# This is the top-level script we give to Kokoro as the entry point for
+# running the "continuous" and "presubmit" jobs.
+
+set -ex
+
+# Change to repo root
+cd $(dirname $0)/../../..
```
Comment (fowles) [inline kokoro/linux/python_aarch64/build.sh:9]: same here

**Review by fowles — COMMENTED** [review]
pzd has been working on java tests lately.  Please wait for his review

File: kokoro/linux/aarch64/python_crosscompile_aarch64.sh (hunk 1 above)
Comment (jtattermusch) [reply kokoro/linux/aarch64/python_crosscompile_aarch64.sh]: It's rebased now (this PR depended on some files introduced by they python PR).

File: kokoro/linux/java_aarch64/build.sh (hunk 2 above)
Comment (jtattermusch) [reply kokoro/linux/java_aarch64/build.sh:9]: Let's leave as is as this is the "standard" way it's currently done in most protobuf's CI testing scripts.
I think it's better to stay consistent with the way it's done in most places in this repository.

Examples:
https://github.com/protocolbuffers/protobuf/blob/3f5fc4df1de8e12b2235c3006593e22d6993c3f5/kokoro/release/python/linux/build_artifacts.sh
https://github.com/protocolbuffers/protobuf/blob/3f5fc4df1de8e12b2235c3006593e22d6993c3f5/kokoro/release/protoc/linux/build.sh
https://github.com/protocolbuffers/protobuf/blob/3f5fc4df1de8e12b2235c3006593e22d6993c3f5/kokoro/linux/benchmark/run.sh

**Review by jtattermusch — COMMENTED** [review]
I rebased the PR after merging the python PR, so now the stuff to review is only java related.

**Review by fowles — APPROVED** [review]

**Comment by jtattermusch** [comment]
@perezd  please review so we can merge this soon.

**Review by perezd — APPROVED** [review]
LGTM!

**Pull request merged**